### PR TITLE
feat: Adding rippable cards pack

### PR DESCRIPTION
**Pull request by EricBartusch**
Added the rippable cards pack.
Still missing two card arts

Rippable cards can be right-clicked to rip into two halves: Art half - Does nothing, costs the original card's cost, and exhausts Text half - Does whatever the original card did, is free, and exhausts.

### CODE REVIEW

File: src/main/java/thePackmaster/cards/rippack/AbstractRippableCard.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    }
+
+    public ArrayList<AbstractCard> getRippedParts() {
+        return rippedParts;
+    }
+
+    public void onRightClick() {
+        if(action == null) {
+            if (canRip()) {
+                CardCrawlGame.sound.play("MAP_CLOSE", 0.0F);
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/rippack/AbstractRippableCard.java:55]: Might be better as an action so it happens at the same time (SFXAction)

File: src/main/java/thePackmaster/cards/rippack/AbstractRippableCard.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                action = new RipCardAction(this, rippedParts.get(0), rippedParts.get(1));
+                att(action);
+                att(new WaitAction(0.1f));
+                att(new WaitAction(0.1f));
+                att(new WaitAction(0.1f));
+                att(new WaitAction(0.1f));
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/rippack/AbstractRippableCard.java:61]: I am not a happy camper

File: src/main/java/thePackmaster/cards/rippack/AbstractRippableCard.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        if (action != null && action.isDone) {
+            action = null;
+        }
+        if (AbstractDungeon.player != null) {
+            clickUpdate();
+        }
+    }
+
+    public boolean canRip() {
+        return AbstractDungeon.player.hand.group.size() != BaseMod.MAX_HAND_SIZE;
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/rippack/AbstractRippableCard.java:94]: CardGroups also have `size()`

File: src/main/java/thePackmaster/cards/rippack/ArtAttackArt.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    public ArtAttackArt() {
+        super(ID, new ArtAttack());
+        baseMagicNumber = magicNumber = 12;
+    }
+
+    @Override
+    public void use(AbstractPlayer p, AbstractMonster m) {
+        AbstractGameEffect rainbowEffect = ArtAttackArtEffect.RainbowBoomerang(m);
+
+        CardCrawlGame.sound.play(makeID("RipPack_Yay"));
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/rippack/ArtAttackArt.java:34]: Could use SFXAction

File: src/main/java/thePackmaster/cards/rippack/SurprisePack.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            return ImageMaster.loadImage(makeImagePath("cards/SurprisePack_u_p.png"));
+        }
+        return super.getPortraitImage();
+    }
+
+    @Override
+    public void upp() {
+        upgradeSecondMagic(1);
+        loadCardImage(makeImagePath("cards/SurprisePack_u.png"));
+        uDesc();
```
Comment (erasels) [inline src/main/java/thePackmaster/cards/rippack/SurprisePack.java:45]: The upgrading description already happens in the abstract, is this needed?

File: src/main/java/thePackmaster/patches/rippack/UnnatePatch.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+            Iterator vacantIterator = copy[0].group.iterator();
+
+            while (vacantIterator.hasNext()) {
+                AbstractCard newCard = (AbstractCard) vacantIterator.next();
+                if (newCard instanceof AbstractPackmasterCard) {
+                    if (((AbstractPackmasterCard) newCard).isUnnate)
+                        placeOnBot.add(newCard);
+                }
+            }
+            if (placeOnBot.size() > 0) {
```
Comment (erasels) [inline src/main/java/thePackmaster/patches/rippack/UnnatePatch.java:33]: Kinda stinky code, I guess you copied this from vacant :^)

File: src/main/resources/anniv5Resources/localization/eng/rippack/Cardstrings.json [hunk 7]
```diff
@@ -0,0 +1,129 @@
+{
+  "${ModID}:DividedFury": {
+    "NAME": "Divided Fury",
+    "DESCRIPTION": "When you ${ModID}:rip, an Attack card, deal !M! damage to ALL enemies. NL When you ${ModID}:rip, a Skill card, gain !M! Block."
```
Comment (erasels) [inline src/main/resources/anniv5Resources/localization/eng/rippack/Cardstrings.json:4]:
```suggestion
    "DESCRIPTION": "When you ${ModID}:rip, an Attack, deal !M! damage to ALL enemies. NL When you ${ModID}:rip, a Skill, gain !M! Block."
```

File: src/main/resources/anniv5Resources/localization/eng/rippack/Keywordstrings.json [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    "NAMES": [
+      "text card",
+      "text_card",
+      "text cards",
+      "text_cards"
+    ],
+    "DESCRIPTION": "The text half of a ripped card. It always exhausts."
+  },
+  {
+    "PROPER_NAME": "Unnate",
```
Comment (erasels) [inline src/main/resources/anniv5Resources/localization/eng/rippack/Keywordstrings.json:32]: My disappointment is immeasurable and my day is ruined.

File: src/main/resources/anniv5Resources/localization/eng/rippack/Keywordstrings.json [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  },
+  {
+    "PROPER_NAME": "Art Card",
+    "NAMES": [
+      "art card",
+      "art_card",
+      "art cards",
+      "art_cards"
+    ],
+    "DESCRIPTION": "The art half of a ripped card. It does nothing and always exhausts."
```
Comment (erasels) [inline src/main/resources/anniv5Resources/localization/eng/rippack/Keywordstrings.json:19]:
```suggestion
    "DESCRIPTION": "The art half of a #yripped card. It does nothing and always #yexhausts."
```

File: src/main/resources/anniv5Resources/localization/eng/rippack/Keywordstrings.json [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+  },
+  {
+    "PROPER_NAME": "Text Card",
+    "NAMES": [
+      "text card",
+      "text_card",
+      "text cards",
+      "text_cards"
+    ],
+    "DESCRIPTION": "The text half of a ripped card. It always exhausts."
```
Comment (erasels) [inline src/main/resources/anniv5Resources/localization/eng/rippack/Keywordstrings.json:29]:
```suggestion
    "DESCRIPTION": "The text half of a #yripped card. It always #yexhausts."
```

File: src/main/resources/anniv5Resources/localization/eng/rippack/Powerstrings.json [hunk 11]
```diff
@@ -0,0 +1,10 @@
+{
+  "${ModID}:DividedFuryPower": {
+    "NAME": "Divided Fury",
+    "DESCRIPTIONS": [
+      "When you rip an Attack card, deal #b",
```
Comment (erasels) [inline src/main/resources/anniv5Resources/localization/eng/rippack/Powerstrings.json:5]:
```suggestion
      "When you #yrip an Attack, deal #b",
```

File: src/main/resources/anniv5Resources/localization/eng/rippack/Powerstrings.json [hunk 12]
```diff
@@ -0,0 +1,10 @@
...
+      " damage to ALL enemies. NL When you rip a Skill card, gain #b",
```
Comment (erasels) [inline src/main/resources/anniv5Resources/localization/eng/rippack/Powerstrings.json:6]:
```suggestion
      " damage to ALL enemies. NL When you #yrip a Skill, gain #b",
```

File: src/main/resources/anniv5Resources/localization/eng/rippack/Powerstrings.json [hunk 13]
```diff
@@ -0,0 +1,10 @@
...
+      " Block."
```
Comment (erasels) [inline src/main/resources/anniv5Resources/localization/eng/rippack/Powerstrings.json:7]:
```suggestion
      " #yBlock."
```

**Review by erasels — CHANGES_REQUESTED** [review]

File: src/main/java/thePackmaster/cards/rippack/AbstractRippableCard.java (hunk 2 above)
Comment (EricBartusch) [reply src/main/java/thePackmaster/cards/rippack/AbstractRippableCard.java:62]: Without these, the two halves of the card go into your hand before the animation of the rip finishes. I could do what I do elsewhere where I force other actions to wait for one to be done before they start

File: src/main/java/thePackmaster/patches/rippack/UnnatePatch.java (hunk 6 above)
Comment (EricBartusch) [reply src/main/java/thePackmaster/patches/rippack/UnnatePatch.java:33]: :sunglasso:

File: src/main/java/thePackmaster/cards/rippack/AbstractRippableCard.java (hunk 2 above)
Comment (erasels) [reply src/main/java/thePackmaster/cards/rippack/AbstractRippableCard.java:62]: Might be better to use a ForceWaitAction if you're scared that people will circumvent the wait with SuperFastMode. (or just instnaciate the action, set its duration to whatever and then queue it.

**Pull request merged**